tests/extras/datasets/pandas/test_sql_dataset.py

```diff
@@ -77,61 +80,67 @@ def test_load_sql_params(self, mocker, table_data_set):
         """Test `load` method invocation"""
         mocker.patch("pandas.read_sql_table")
         table_data_set.load()
-        self._assert_pd_called_once()
+        pd.read_sql_table.assert_called_once_with(
+            table_name=TABLE_NAME, con=table_data_set.engines[CONNECTION]
+        )
 
-    def test_load_driver_missing(self, mocker, table_data_set):
+    def test_load_driver_missing(self, mocker):
         """Check the error when the sql driver is missing"""
         mocker.patch(
-            "pandas.read_sql_table",
+            "kedro.extras.datasets.pandas.sql_dataset.create_engine",
             side_effect=ImportError("No module named 'mysqldb'"),
         )
         with pytest.raises(DataSetError, match=ERROR_PREFIX + "mysqlclient"):
-            table_data_set.load()
-        self._assert_pd_called_once()
+            SQLTableDataSet(table_name=TABLE_NAME, credentials=dict(con=CONNECTION))
 
-    def test_invalid_module(self, mocker, table_data_set):
+    def test_invalid_module(self, mocker):
         """Test that if an invalid module/driver is encountered by SQLAlchemy
         then the error should contain the original error message"""
         _err = ImportError("Invalid module some_module")
-        mocker.patch("pandas.read_sql_table", side_effect=_err)
+        mocker.patch(
+            "kedro.extras.datasets.pandas.sql_dataset.create_engine", side_effect=_err
+        )
         pattern = ERROR_PREFIX + r"Invalid module some\_module"
         with pytest.raises(DataSetError, match=pattern):
-            table_data_set.load()
-        self._assert_pd_called_once()
+            SQLTableDataSet(table_name=TABLE_NAME, credentials=dict(con=CONNECTION))
 
-    def test_load_unknown_module(self, mocker, table_data_set):
+    def test_load_unknown_module(self, mocker):
         """Test that if an unknown module/driver is encountered by SQLAlchemy
         then the error should contain the original error message"""
         mocker.patch(
-            "pandas.read_sql_table",
+            "kedro.extras.datasets.pandas.sql_dataset.create_engine",
             side_effect=ImportError("No module named 'unknown_module'"),
         )
         pattern = ERROR_PREFIX + r"No module named \'unknown\_module\'"
         with pytest.raises(DataSetError, match=pattern):
-            table_data_set.load()
+            SQLTableDataSet(table_name=TABLE_NAME, credentials=dict(con=CONNECTION))
```

Comment:
Some of these tests are now duplicated, because we're checking for errors at instantiation, rather than on `load` AND `save`. TODO: Deduplicate these and move under `TestSQLTableDataSet`.